alvr/client_openxr/src/stream.rs

```diff
@@ -628,6 +629,20 @@ fn stream_input_loop(
             device_motions.append(&mut interaction::get_bd_motion_trackers(source, now));
         }
 
+        let button_entries = interaction::update_buttons(&xr_session, &int_ctx.button_actions);
+        if !button_entries.is_empty() {
+            core_ctx.send_buttons(button_entries);
+        }
```

Comment:
You checked for any potential consequences this might have? I'm assuming that the button update being earlier won't matter, but if you aren't fully sure that the point where the buttons are sent doesn't matter, then I'd encourage you to just move the sending part after the `send_tracking` again.
(Also I'm pretty sure if you manually drop it wont drop until the manual drop so you'd be doing a whole sending operation whilst the lock is being held which could also be avoided.)

Reply:
I don't think timing matters. But following your advice I might just relock the interaction context and send the buttons after, just to keep a separation of concern in the code flow